src/system/SystemLayer.cpp

```diff
@@ -31,5 +34,16 @@ CHIP_ERROR Layer::ScheduleLambdaBridge(LambdaBridge && bridge)
     return lReturn;
 }
 
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
+CHIP_ERROR Layer::RunWithMatterContextLock(std::function<CHIP_ERROR()> nonBlockingFunc)
```

Comment:
how about placing this in some separate CPP file and in that case we can make the link decision in build.gn? Something like `SystemLayer_Locking.cpp`.

Reply:
moved this function to the header now